spec/civildate.html

```diff
@@ -1,164 +1,566 @@
+            1. Throw a *TypeError* exception.
+          1. Let _value_ be ? TemporalToInteger().
+          1. Set _result_'s internal slot whose name is the Internal Slot value of the current row to _value_.
+        1. Return _result_.
+      </emu-alg>
+
+    <emu-clause id=sec-temporal-ToPartialDate aoid=ToPartialDate>
+      <h1>ToPartialDate ( _date_ )</h1>
+      <emu-alg>
+        1. Let _result_ be {
+          [[Year]]: *undefined*,
+          [[Month]]: *undefined*,
+          [[Day]]: *undefined*,
+          }.
+        1. If Type(_date_) is Undefined or Null, then
+          1. Return _result_.
+        1. <mark>TODO: Handle strings?</mark>
+        1. If Type(_date_) is not Object, then
+          1. Throw a *TypeError* exception.
+        1. For each row of <emu-xref href="#table-temporal-datelike-properties"></emu-xref>, except the header row, in table order, do
+          1. Let _property_ be the Property value of the current row.
+          1. Let _value_ be ? TemporalToInteger(? Get(_date_, _property_)).
+          1. Set _result_'s internal slot whose name is the Internal Slot value of the current row to _value_.
+        1. Return _result_.
+      </emu-alg>
+
+      <emu-table id="table-temporal-datelike-properties">
+        <emu-caption>Properties of a DateLike</emu-caption>
+        <table class="real-table">
+          <thead>
+            <tr>
+              <th>Internal Slot</th>
+              <th>Property</th>
+            </tr>
+          </thead>
+          <tbody>
+            <tr>
+              <td>[[Day]]</td>
+              <td>`"day"`</td>
+            </tr>
+            <tr>
+              <td>[[Month]]</td>
+              <td>`"month"`</td>
+            </tr>
+            <tr>
+              <td>[[Year]]</td>
+              <td>`"year"`</td>
+            </tr>
+          </tbody>
+        </table>
+      </emu-table>
+    </emu-clause>
+
+    <emu-clause id=sec-temporal-RegulateDate aoid=RegulateDate>
+      <h1>RegulateDate ( _y_, _m_, _d_, _disambiguation_ )</h1>
+      <emu-alg>
+        1. <mark>TODO: consult _disambiguation_.</mark>
+        1. If ! ValidateDate(_y_, _m_, _d_) is *false*, then
+          1. Throw a *RangeError* exception.
+      </emu-alg>
+    </emu-clause>
+
+    <emu-clause id=sec-temporal-ValidateDate aoid=ValidateDate>
+      <h1>ValidateDate ( _y_, _m_, _d_ )</h1>
+      <emu-alg>
+        1. Assert: _y_, _m_, and _d_ are integer Number values.
+        1. If _m_ &lt; 1 or _m_ &gt; 12, then
+          1. Return *false*.
+        1. Let _maxDay_ be DaysInMonth(_y_, _m_).
+        1. If _d_ &lt; 1 or _d_ &gt; _maxDay_, then
+          1. Return *false*.
+        1. Return *true*.
+      </emu-alg>
+    </emu-clause>
+
+    <emu-clause id=sec-temporal-PadYear aoid=PadYear>
+      <h1>PadYear ( _y_ )</h1>
+      <emu-alg>
+        1. Assert: _y_ is an integer Number value.
+        1. If _y_ &gt; 999 and _y_ &lt; 9999, then
+          1. <mark>TODO: Why is this different?</mark>
```

Comment:
smaller/greater years are padded to 6 digits and always have the +/- sign in front